### PR TITLE
Fix action alias update API endpoint

### PR DESCRIPTION
**Pull request by Kami**
This fixes bug reported in https://github.com/StackStorm/st2/issues/3170.

It looks like this regression was introduced as part of https://github.com/StackStorm/st2/pull/2841, but we didn't have any tests for this API endpoint so the regression wasn't caught earlier.

### CODE REVIEW

**Comment by m4dcoder** [comment]
It's not obvious from the original problem posted at https://github.com/StackStorm/st2/issues/3170 to the fix here.  Maybe more explanation in the summary will help.  Otherwise the code LGTM.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -49,6 +49,7 @@ in development
   put the workflow in a PAUSED state in mistral. (improvement)
 * Add support for evaluating jinja expressions in mistral workflow definition where yaql
   expressions are typically accepted. (improvement)
+* Fix action alias update API endpoint. (bug fox)
```
Comment (m4dcoder) [inline CHANGELOG.rst]: Do you mean bug fix?

Comment (humblearner) [inline CHANGELOG.rst]: minor: s/fox/fix

File: st2api/st2api/controllers/v1/actionalias.py [hunk 2]
```diff
@@ -156,13 +156,16 @@ def post(self, action_alias):
 
         return action_alias_api
 
-    @request_user_has_resource_db_permission(permission_type=PermissionType.ACTION_MODIFY)
+    @request_user_has_resource_db_permission(permission_type=PermissionType.ACTION_ALIAS_MODIFY)
     @jsexpose(arg_types=[str], body_cls=ActionAliasAPI)
-    def put(self, action_alias_ref_or_id, action_alias):
+    def put(self, action_alias, action_alias_ref_or_id):
```
Comment (lakshmi-kannan) [inline st2api/st2api/controllers/v1/actionalias.py]: Interesting. I didn't realize body is first arg and ref is second.

File: CHANGELOG.rst (hunk 1 above)
Comment (Kami) [inline CHANGELOG.rst]: lol, yeah, good catch :D

**Comment by Kami** [comment]
> It's not obvious from the original problem posted at #3170 to the fix here. Maybe more explanation in the summary will help. Otherwise the code LGTM.

@m4dcoder Yeah, regression / bug was introduced as part of - https://github.com/StackStorm/st2/pull/2841

That PR changed the arguments order in the method signature and the fix is changing the arguments order in the method signature (previously first argument was id, now it's the actual request payload).

File: st2api/st2api/controllers/v1/actionalias.py (hunk 2 above)
Comment (Kami) [inline st2api/st2api/controllers/v1/actionalias.py]: Yeah, I was personally not a fan of that change and I prefer id to be first (this way it's consistent with jsexpose and everything and less magic, etc.), but @enykeev some how managed to continence me in https://github.com/StackStorm/st2/pull/2841 that this is good idea :P

Comment (enykeev) [inline st2api/st2api/controllers/v1/actionalias.py]: In st2api v2, position of the argument would not matter at all, all of them will be passed as kwargs.

**Pull request merged**